subsys/bluetooth/mesh/shell/shell_sensor_cli.c

```diff
@@ -52,22 +54,27 @@ static int cmd_desc_get(const struct shell *shell, size_t argc, char *argv[])
 		uint32_t count = CONFIG_BT_MESH_SHELL_SENSOR_CLI_MAX_SENSORS;
 		struct bt_mesh_sensor_info rsp[CONFIG_BT_MESH_SHELL_SENSOR_CLI_MAX_SENSORS];
 
-		int err = bt_mesh_sensor_cli_desc_all_get(cli, NULL, rsp, &count);
+		err = bt_mesh_sensor_cli_desc_all_get(cli, NULL, rsp, &count);
 
 		descriptors_print(shell, err, rsp, count);
 		return err;
 	}
 
 	struct bt_mesh_sensor_descriptor rsp;
-	uint32_t sensor_id = (uint16_t)strtol(argv[1], NULL, 0);
+	uint32_t sensor_id = shell_strtoul(argv[1], 10, &err);
```

Comment:
This absolutely needs hex support, I've never worked with decimal sensor IDs. Like earlier comments, would be nice with base 0 if possible.